hail/src/main/scala/is/hail/backend/HailTaskContext.scala

```diff
@@ -1,11 +1,33 @@
 package is.hail.backend
 
+import is.hail.HailContext
+import is.hail.annotations.RegionPool
+import is.hail.backend.local.{LocalBackend, LocalTaskContext}
+import is.hail.backend.spark.{SparkBackend, SparkTaskContext}
+import org.apache.spark.TaskContext
+
 object HailTaskContext {
   def get(): HailTaskContext = taskContext.get
 
-  private[this] val taskContext: ThreadLocal[HailTaskContext] = new ThreadLocal[HailTaskContext]
-  protected[backend] def setTaskContext(tc: HailTaskContext): Unit = taskContext.set(tc)
-  protected[backend] def unset(): Unit = taskContext.remove()
+  private[this] val taskContext: ThreadLocal[HailTaskContext] = new ThreadLocal[HailTaskContext]() {
+    override def initialValue(): HailTaskContext = {
+      val backend = HailContext.get.backend
```

Comment:
It occurs to me that this is going to be a problem. Can't decide how to construct based on the backend, because you can't ask for the `HailContext` on a worker. Not sure what the best way to do this.

Reply:
oh, I think just assume it's spark backend. This can just be the lines 16-18 I think -- the other backends will always be setting the task context in parallelizeAndComputeWithIndex

Reply:
How does this work in the `ServiceBackend`? I'm not sure where the task context will be set.